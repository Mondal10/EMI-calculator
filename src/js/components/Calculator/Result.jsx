import React from 'react';

import styled from 'styled-components';

import PieChart from  '../Calculator/PieChart';

const ResultContainer = styled.div`
  display: flex;
  font-size: 20px;
  padding: 2%;
  border-radius: 20px;
  color: #3E4C5F;
  justify-content: space-between;
  background-color: #ededf2;
`;

const ValuesContainer = styled.div`
  width: 50%;
`;

const Value = styled.div`
  padding: 2% 0;
`;

const ChartContainer = styled.div`
  width: 50%;
  padding: 2%;
  border-radius: 20px;
  background-color: #d1e0f5;
`;

function Result({ resultObj, amount }) {
  const { loanEMI, totalInterestToPay, totalPayment } = resultObj;

  // % calculation of Principal Loan Amount and Total Interest
  const principalLoanAmt = parseFloat(((amount / totalPayment) * 100).toPrecision(4));
  const totalInterest = 100 - principalLoanAmt;

  const rupeeFormatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'INR',
    minimumFractionDigits: 0
  })

  return (
    <ResultContainer>
      <ValuesContainer>
        <Value>
          Loan EMI: {loanEMI ? rupeeFormatter.format(loanEMI) : null}
        </Value>
        <Value>
          Total Interest Payable: {totalInterestToPay ? rupeeFormatter.format(totalInterestToPay) : null}
        </Value>
        <Value>
          Total Payment: {totalPayment ? rupeeFormatter.format(totalPayment) : null}
        </Value>
      </ValuesContainer>
      <ChartContainer>
        <PieChart principalLoanAmt={principalLoanAmt} totalInterest={totalInterest}/>
      </ChartContainer>
    </ResultContainer>
  )
}

export default Result;
