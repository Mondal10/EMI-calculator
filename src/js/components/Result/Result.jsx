import React from 'react';

import styled from 'styled-components';

import PieChart from '../PieChart/PieChart';

const ResultContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  font-size: 20px;
  padding: 2%;
  border-radius: 20px;
  color: #232627;
  justify-content: space-between;
  background-color: #ededf2;
`;

const ValuesContainer = styled.div`
  width: auto;
`;

const Value = styled.div`
  padding: 2% 0;
`;

const ChartContainer = styled.div`
  width: auto;
  padding: 2%;
  margin: 20px auto;
  border-radius: 20px;
  background-color: #232627;
`;

function Result({ resultObj, amount }) {
  const { loanEMI, totalInterestToPay, totalPayment } = resultObj;

  // % calculation of Principal Loan Amount and Total Interest
  const principalLoanAmt = parseFloat(((amount / totalPayment) * 100).toPrecision(4));
  const totalInterest = 100 - principalLoanAmt;

  // Convert to INR format with symbol
  const rupeeFormatter = new Intl.NumberFormat('en-IN', {
    style: 'currency',
    currency: 'INR',
    minimumFractionDigits: 0
  })

  return (
    <ResultContainer>
      <ValuesContainer>
        <Value id="loanEmi">
          Loan EMI: {loanEMI ? rupeeFormatter.format(loanEMI) : null}
        </Value>
        <Value id="totalInterest">
          Total Interest Payable: {totalInterestToPay ? rupeeFormatter.format(totalInterestToPay) : null}
        </Value>
        <Value id="totalPayment">
          Total Payment: {totalPayment ? rupeeFormatter.format(totalPayment) : null}
        </Value>
      </ValuesContainer>
      <ChartContainer>
        <PieChart principalLoanAmt={principalLoanAmt} totalInterest={totalInterest} />
      </ChartContainer>
    </ResultContainer>
  )
}

export default Result;
