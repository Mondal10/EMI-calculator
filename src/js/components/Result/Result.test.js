import React from 'react';
import renderer from 'react-test-renderer';
import Result from './Result';

describe("Result Snapshot", () => {
  const loanAmount = 1000000;
  const resultObj = {
    loanEMI: 13493,
    totalInterestToPay: 619220,
    totalPayment: 1619220,
  };

  it('Result rendered successfully', () => {
    const tree = renderer.create(<Result resultObj={resultObj} amount={loanAmount} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
