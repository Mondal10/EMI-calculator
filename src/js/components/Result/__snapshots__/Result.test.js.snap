// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Snapshot Result rendered successfully 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 20px;
  padding: 2%;
  border-radius: 20px;
  color: #232627;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #ededf2;
}

.c1 {
  width: auto;
}

.c2 {
  padding: 2% 0;
}

.c3 {
  width: auto;
  padding: 2%;
  margin: 20px auto;
  border-radius: 20px;
  background-color: #232627;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      id="loanEmi"
    >
      Loan EMI: 
      ₹ 13,493
    </div>
    <div
      className="c2"
      id="totalInterest"
    >
      Total Interest Payable: 
      ₹ 6,19,220
    </div>
    <div
      className="c2"
      id="totalPayment"
    >
      Total Payment: 
      ₹ 16,19,220
    </div>
  </div>
  <div
    className="c3"
  />
</div>
`;
